x-pack/plugin/src/test/resources/rest-api-spec/test/runtime_fields/10_keyword.yml

```diff
@@ -138,6 +139,31 @@ setup:
   - match: {aggregations.dow.buckets.1.key: Monday}
   - match: {aggregations.dow.buckets.1.doc_count: 1}
 
+  # Update the mapping and make sure the cache doesn't still have the old results
+  - do:
+      indices.put_mapping:
+        index: sensor
+        body:
+          runtime:
+            day_of_week:
+              type: keyword
+              script: |
+                emit(doc['timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.SHORT, Locale.ROOT));
+  - do:
+      search:
+        index: sensor
+        body:
+          size: 0
+          aggs:
+            dow:
+              terms:
+                field: day_of_week
+  - match: {hits.total.value: 6}
+  - match: {aggregations.dow.buckets.0.key: Fri}
+  - match: {aggregations.dow.buckets.0.doc_count: 1}
+  - match: {aggregations.dow.buckets.1.key: Mon}
+  - match: {aggregations.dow.buckets.1.doc_count: 1}
+
```

Comment:
This is the very first thing I wrote in the PR, just to make sure I could be sure it'd fail. And it sure did!